File: backend/src/main/java/com/devsuperior/movieflix/entities/Movie.java
package com.devsuperior.movieflix.entities;


import org.springframework.data.repository.cdi.Eager;

import javax.persistence.*;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "movie")
public class Movie implements Serializable {
    private static final long serialVersionUID =1L;
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    private String subTitle;
    private Integer year;
    @Column(name = "img_url")
    private String imgUrl;
    private String synopsis;
    @ManyToOne
    @JoinColumn(foreignKey = @ForeignKey(name = "FK_Movie_genreId"))
    private Genre genre;

    @OneToMany(mappedBy = "movie", fetch = FetchType.EAGER)
    private List<Review> reviews = new ArrayList<>();

    public Movie() {
    }

    public static long getSerialVersionUID() {
        return serialVersionUID;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getSubTitle() {
        return subTitle;
    }

    public void setSubTitle(String subTitle) {
        this.subTitle = subTitle;
    }

    public Integer getYear() {
        return year;
    }

    public void setYear(Integer year) {
        this.year = year;
    }

    public String getImgUrl() {
        return imgUrl;
    }

    public void setImgUrl(String imgUrl) {
        this.imgUrl = imgUrl;
    }

    public String getSynopsis() {
        return synopsis;
    }

    public void setSynopsis(String synopsis) {
        this.synopsis = synopsis;
    }

    public Genre getGenre() {
        return genre;
    }

    public void setGenre(Genre genre) {
        this.genre = genre;
    }

    public List<Review> getReviews() {
        return reviews;
    }

    public void setReviews(List<Review> reviews) {
        this.reviews = reviews;
    }
}

File: front-web/src/core/types/Movie.ts

export type MovieResponse = {
    content: Movie[];
    totalPages: number;
}

export type Movie  = {
    id: number;
    title: string;
    subTitle: string;
    year: number;
    imgUrl: string;
    synopsis: string;
    genre: Genre;
    reviews: Review[];
}

export type Review = {
    id: number;
    text: string;
    user: User;
    movie: Movie;
}

export type Genre = {
    id: number;
    name: string;
}

export type User = {
    id: number;
    userName?: string;
    email?: string;
    password?: string;
    roles?: Role[];
    reviews?: Review[];
}

export type Role = {
    id: number;
    authority: string;
}

export type GenreResponse = {
    content: Genre[];
    totalPages:number;
}

export type ReviewResponse = {
    content: Review[];
    totalPages:number;
}